plugin/documents.py

```diff
@@ -216,6 +265,95 @@ def on_hover(self, point: int, hover_zone: int) -> None:
             return
         self.view.run_command("lsp_hover", {"point": point})
 
+    # --- textDocument/signatureHelp -----------------------------------------------------------------------------------
+
+    def _do_signature_help(self) -> None:
+        # NOTE: We take the beginning of the region to check the previous char (see last_char variable). This is for
+        # when a language server inserts a snippet completion.
+        pos = self._stored_region.a
+        if pos == -1:
+            return
+        if not self.view.match_selector(pos, self.view.settings().get("auto_complete_selector") or ""):  # ???
+            return
+        provider = None  # type: Optional[Dict[str, Any]]
+        session = None  # type: Optional[Session]
+        for sv in self.session_views_async():
+            provider = sv.session.get_capability("signatureHelpProvider")
+            if provider:
+                session = sv.session
+                break
+        if not provider or not session:
+            return
+        triggers = provider.get("triggerCharacters") or []
```

Comment:
I believe this block could be simplified to:
```py
        session = self.session("signatureHelpProvider")
        if not session:
            return
        triggers = session.get_capability("signatureHelpProvider.triggerCharacters") or []
```
Since we are handling only one session anyway.